File: resources/expected/ros2/datatype-examples/lax/src/datatypes_system_cpp_pkg/src/user_code/producer_producer_src.cpp
#include "datatypes_system_cpp_pkg/user_headers/producer_producer_src.hpp"
#include "datatypes_system_cpp_pkg/base_headers/enum_converter.hpp"

//===========================================================
// This file will not be overwritten when re-running Codegen
//===========================================================

//=================================================
//  I n i t i a l i z e    E n t r y    P o i n t
//=================================================
void producer_producer::initialize()
{
    PRINT_INFO("Initialize Entry Point invoked");

    // Initialize the node
}

//=================================================
//  C o m p u t e    E n t r y    P o i n t
//=================================================
void producer_producer::timeTriggered()
{
    // Handle communication

    // Example publishing messages
    datatypes_system_cpp_pkg_interfaces::msg::Boolean myBoolean = datatypes_system_cpp_pkg_interfaces::msg::Boolean();
    put_myBoolean(myBoolean);
    datatypes_system_cpp_pkg_interfaces::msg::Integer64 myInteger = datatypes_system_cpp_pkg_interfaces::msg::Integer64();
    put_myInteger(myInteger);
    datatypes_system_cpp_pkg_interfaces::msg::Float64 myFloat = datatypes_system_cpp_pkg_interfaces::msg::Float64();
    put_myFloat(myFloat);
    datatypes_system_cpp_pkg_interfaces::msg::Character myCharacter = datatypes_system_cpp_pkg_interfaces::msg::Character();
    put_myCharacter(myCharacter);
    datatypes_system_cpp_pkg_interfaces::msg::String myString = datatypes_system_cpp_pkg_interfaces::msg::String();
    put_myString(myString);
    datatypes_system_cpp_pkg_interfaces::msg::Integer8 myInt8 = datatypes_system_cpp_pkg_interfaces::msg::Integer8();
    put_myInt8(myInt8);
    datatypes_system_cpp_pkg_interfaces::msg::Integer16 myInt16 = datatypes_system_cpp_pkg_interfaces::msg::Integer16();
    put_myInt16(myInt16);
    datatypes_system_cpp_pkg_interfaces::msg::Integer32 myInt32 = datatypes_system_cpp_pkg_interfaces::msg::Integer32();
    put_myInt32(myInt32);
    datatypes_system_cpp_pkg_interfaces::msg::Integer64 myInt64 = datatypes_system_cpp_pkg_interfaces::msg::Integer64();
    put_myInt64(myInt64);
    datatypes_system_cpp_pkg_interfaces::msg::Unsigned8 myUInt8 = datatypes_system_cpp_pkg_interfaces::msg::Unsigned8();
    put_myUInt8(myUInt8);
    datatypes_system_cpp_pkg_interfaces::msg::Unsigned16 myUInt16 = datatypes_system_cpp_pkg_interfaces::msg::Unsigned16();
    put_myUInt16(myUInt16);
    datatypes_system_cpp_pkg_interfaces::msg::Unsigned32 myUInt32 = datatypes_system_cpp_pkg_interfaces::msg::Unsigned32();
    put_myUInt32(myUInt32);
    datatypes_system_cpp_pkg_interfaces::msg::Unsigned64 myUInt64 = datatypes_system_cpp_pkg_interfaces::msg::Unsigned64();
    put_myUInt64(myUInt64);
    datatypes_system_cpp_pkg_interfaces::msg::Float32 myFloat32 = datatypes_system_cpp_pkg_interfaces::msg::Float32();
    put_myFloat32(myFloat32);
    datatypes_system_cpp_pkg_interfaces::msg::Float64 myFloat64 = datatypes_system_cpp_pkg_interfaces::msg::Float64();
    put_myFloat64(myFloat64);
    datatypes_system_cpp_pkg_interfaces::msg::MyEnum myEnum = datatypes_system_cpp_pkg_interfaces::msg::MyEnum();
    put_myEnum(myEnum);
    datatypes_system_cpp_pkg_interfaces::msg::MyStructi myStruct = datatypes_system_cpp_pkg_interfaces::msg::MyStructi();
    put_myStruct(myStruct);
    datatypes_system_cpp_pkg_interfaces::msg::MyArrayOneDim myArray1 = datatypes_system_cpp_pkg_interfaces::msg::MyArrayOneDim();
    put_myArray1(myArray1);
    datatypes_system_cpp_pkg_interfaces::msg::MyArrayUnbounded myArray2 = datatypes_system_cpp_pkg_interfaces::msg::MyArrayUnbounded();
    put_myArray2(myArray2);
    datatypes_system_cpp_pkg_interfaces::msg::MyArrayTwoDim myArray3 = datatypes_system_cpp_pkg_interfaces::msg::MyArrayTwoDim();
    put_myArray3(myArray3);
}

File: resources/expected/ros2/python-datatype-examples/lax/src/datatypes_system_py_pkg/datatypes_system_py_pkg/user_code/producer_producer_src.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from rosidl_runtime_py.convert import message_to_yaml
from datatypes_system_py_pkg.base_code.producer_producer_base_src import producer_producer_base
from datatypes_system_py_pkg_interfaces.msg import Boolean
from datatypes_system_py_pkg_interfaces.msg import Integer64
from datatypes_system_py_pkg_interfaces.msg import Float64
from datatypes_system_py_pkg_interfaces.msg import Character
from datatypes_system_py_pkg_interfaces.msg import String
from datatypes_system_py_pkg_interfaces.msg import Integer8
from datatypes_system_py_pkg_interfaces.msg import Integer16
from datatypes_system_py_pkg_interfaces.msg import Integer32
from datatypes_system_py_pkg_interfaces.msg import Unsigned8
from datatypes_system_py_pkg_interfaces.msg import Unsigned16
from datatypes_system_py_pkg_interfaces.msg import Unsigned32
from datatypes_system_py_pkg_interfaces.msg import Unsigned64
from datatypes_system_py_pkg_interfaces.msg import Float32
from datatypes_system_py_pkg_interfaces.msg import MyEnum
from datatypes_system_py_pkg_interfaces.msg import MyStructi
from datatypes_system_py_pkg_interfaces.msg import MyArrayOneDim
from datatypes_system_py_pkg_interfaces.msg import MyArrayUnbounded
from datatypes_system_py_pkg.base_code.enum_converter import *

#===========================================================
# This file will not be overwritten when re-running Codegen
#===========================================================
class producer_producer(producer_producer_base):
    def __init__(self):
        super().__init__()
        # invoke initialize entry point
        self.initialize()

        self.get_logger().info("producer_producer infrastructure set up")

#=================================================
#  I n i t i a l i z e    E n t r y    P o i n t
#=================================================
    def initialize(self):
        self.get_logger().info("Initialize Entry Point invoked")

        # Initialize the node

        # Initialize the node's incoming data port values here


#=================================================
#  C o m p u t e    E n t r y    P o i n t
#=================================================
    def message_to_string(self, msg):
        yaml_str = message_to_yaml(msg)
        return yaml_str

    def timeTriggered(self):
        # Handle communication

        # Example publishing messages
        myBoolean = Boolean()
        self.put_myBoolean(myBoolean)

        myInteger = Integer64()
        self.put_myInteger(myInteger)

        myFloat = Float64()
        self.put_myFloat(myFloat)

        myCharacter = Character()
        self.put_myCharacter(myCharacter)

        myString = String()
        self.put_myString(myString)

        myInt8 = Integer8()
        self.put_myInt8(myInt8)

        myInt16 = Integer16()
        self.put_myInt16(myInt16)

        myInt32 = Integer32()
        self.put_myInt32(myInt32)

        myInt64 = Integer64()
        self.put_myInt64(myInt64)

        myUInt8 = Unsigned8()
        self.put_myUInt8(myUInt8)

        myUInt16 = Unsigned16()
        self.put_myUInt16(myUInt16)

        myUInt32 = Unsigned32()
        self.put_myUInt32(myUInt32)

        myUInt64 = Unsigned64()
        self.put_myUInt64(myUInt64)

        myFloat32 = Float32()
        self.put_myFloat32(myFloat32)

        myFloat64 = Float64()
        self.put_myFloat64(myFloat64)

        myEnum = MyEnum()
        self.put_myEnum(myEnum)

        myStruct = MyStructi()
        self.put_myStruct(myStruct)

        myArray1 = MyArrayOneDim()
        self.put_myArray1(myArray1)

        myArray2 = MyArrayUnbounded()
        self.put_myArray2(myArray2)


#=================================================
#  Include any additional declarations here
#=================================================
# Additions within these tags will be preserved when re-running Codegen

# Additions within these tags will be preserved when re-running Codegen
